src/deployment/deploylib/configuration.py

```diff
@@ -78,8 +78,7 @@ def parse_nsg_json(self, config: Any) -> None:
                 "Empty Inner Configuration File Provided. Please Provide Valid Config."
             )
         if (
-            None in proxy_config.keys()
-            or "allowed_ips" not in proxy_config.keys()
+            "allowed_ips" not in proxy_config.keys()
```

Comment:
This is done again a few lines above on line 67 and needs to be removed, as well.

Reply:
Also, with Python dicts, `k in d` checks if a value `k` occurs as a _key_ in a dict `d`.

Example:
```python
>>> d = {'k': 'v'}
>>> 'k' in d
True
>>> 'v' in d
False
```
So the calls to `keys()` are unnecessary.